### PR TITLE
tests: CI of updating from master > this commit.

### PR DESCRIPTION
**Pull request by JohnMcLear**
In response to cypress-eslint breaking changes I thought I'd put some CI testing for if a PR might break automated upgrading.

Matrix usage is probably overkill.  Cypress might be too but gives us decent coverage.

Needs squashing.

### CODE REVIEW

File: .github/workflows/major-version-git-pull-update.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      run: src/bin/installDeps.sh
+
+    - name: Run the backend tests
+      run: cd src && npm test
+
+    - name: Git fetch
+      run: git fetch
+
+    - name: Checkout this branch over master
+      run: git checkout ${GITHUB_REF##*/}
```
Comment (rhansen) [inline .github/workflows/major-version-git-pull-update.yml:73]:
```suggestion
      run: git checkout "${GITHUB_SHA}"
```

File: .github/workflows/major-version-git-pull-update.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    - name: Pull
+      run: git pull
+
```
Comment (rhansen) [inline .github/workflows/major-version-git-pull-update.yml:77]: These lines should be unnecessary.

```suggestion
```

File: .github/workflows/major-version-git-pull-update.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    - uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node }}
+
+    - name: Install libreoffice
+      run: |
+        sudo add-apt-repository -y ppa:libreoffice/ppa
+        sudo apt update
+        sudo apt install -y --no-install-recommends libreoffice libreoffice-pdfimport
+
```
Comment (rhansen) [inline .github/workflows/major-version-git-pull-update.yml:36]: `settings.json` doesn't enable soffice import/export by default so delete these lines:

```suggestion
```

**Comment by JohnMcLear** [comment]
tnx @rhansen

**Pull request merged**